### PR TITLE
MULE-16174: Refactor of ProcessingStrategies

### PR DESCRIPTION
**Pull request by elrodro83**
* Move work queue specific stuff to work queue code.
* Cleanup unused parameters in PS construction

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/WorkQueueStreamProcessingStrategyFactory.java [hunk 1]
```diff
@@ -87,6 +117,43 @@ protected WorkQueueStreamProcessingStrategy(Supplier<Scheduler> ringBufferSchedu
...
+      final long shutdownTimeout = flowConstruct.getMuleContext().getConfiguration().getShutdownTimeout();
+      WorkQueueProcessor<EventWrapper> processor =
+          WorkQueueProcessor.<EventWrapper>builder().executor(ringBufferSchedulerSupplier.get()).bufferSize(bufferSize)
+              .waitStrategy(waitStrategy.getReactorWaitStrategy()).build();
+      int subscriberCount = maxConcurrency < subscribers ? maxConcurrency : subscribers;
+      CountDownLatch completionLatch = new CountDownLatch(subscriberCount);
+      for (int i = 0; i < subscriberCount; i++) {
+        processor
+            .doOnSubscribe(subscription -> currentThread().setContextClassLoader(executionClassloader))
+            .map(ew -> ew.getWrappedEvent())
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/WorkQueueStreamProcessingStrategyFactory.java]: Code Smell Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2wOotIXnmsUMK6V&open=AWkQk2wOotIXnmsUMK6V)

[ISSUE_KEY]: AWkQk2wOotIXnmsUMK6V

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/WorkQueueStreamProcessingStrategyFactory.java [hunk 2]
```diff
@@ -133,4 +200,46 @@ public void stop() throws MuleException {
 
   }
 
+  private static final class EventWrapper {
+
+    CoreEvent wrappedEvent;
+
+    public EventWrapper(CoreEvent event) {
+      this.wrappedEvent = event;
+      ((BaseEventContext) (wrappedEvent.getContext())).getRootContext().onTerminated((e, t) -> {
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/WorkQueueStreamProcessingStrategyFactory.java]: Code Smell Code Smell: Remove useless curly braces around statement ([squid:S1602](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1602))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2wOotIXnmsUMK6U&open=AWkQk2wOotIXnmsUMK6U)

[ISSUE_KEY]: AWkQk2wOotIXnmsUMK6U

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractReactorStreamProcessingStrategy.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                                          Supplier<Scheduler> cpuLightSchedulerSupplier, int parallelism,
+                                          int maxConcurrency, boolean maxConcurrencyEagerCheck) {
+    super(subscribers, maxConcurrency, maxConcurrencyEagerCheck);
+    this.cpuLightSchedulerSupplier = cpuLightSchedulerSupplier;
+    this.parallelism = parallelism;
+  }
+
+  @Override
+  public ReactiveProcessor onProcessor(ReactiveProcessor processor) {
+    reactor.core.scheduler.Scheduler cpuLightScheduler = fromExecutorService(decorateScheduler(getCpuLightScheduler()));
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractReactorStreamProcessingStrategy.java]: Code Smell Code Smell: Rename "cpuLightScheduler" which hides the field declared at line 25. ([squid:HiddenFieldCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AHiddenFieldCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2v5otIXnmsUMK6T&open=AWkQk2v5otIXnmsUMK6T)

[ISSUE_KEY]: AWkQk2v5otIXnmsUMK6T

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamWorkQueueProcessingStrategyFactory.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  public void setWaitStrategy(String waitStrategy) {
+    this.waitStrategy = waitStrategy;
+  }
+
+  protected String getWaitStrategy() {
+    return waitStrategy;
+  }
+
+  protected Supplier<Scheduler> getRingBufferSchedulerSupplier(MuleContext muleContext, String schedulersNamePrefix) {
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamWorkQueueProcessingStrategyFactory.java]: Code Smell Code Smell: Remove this use of "MuleContext"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2vfotIXnmsUMK6S&open=AWkQk2vfotIXnmsUMK6S)

[ISSUE_KEY]: AWkQk2vfotIXnmsUMK6S

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamWorkQueueProcessingStrategyFactory.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ * transaction by default.
+ *
+ * @since 4.2
+ */
+public abstract class AbstractStreamWorkQueueProcessingStrategyFactory extends AbstractStreamProcessingStrategyFactory {
+
+  protected static final String DEFAULT_WAIT_STRATEGY =
+      getProperty(SYSTEM_PROPERTY_PREFIX + "DEFAULT_WAIT_STRATEGY", LITE_BLOCKING.name());
+
+  protected static String RING_BUFFER_SCHEDULER_NAME_SUFFIX = ".ring-buffer";
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamWorkQueueProcessingStrategyFactory.java]: Code Smell Code Smell: Rename this field "RING_BUFFER_SCHEDULER_NAME_SUFFIX" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. ([squid:S3008](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3008))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2vfotIXnmsUMK6R&open=AWkQk2vfotIXnmsUMK6R)

[ISSUE_KEY]: AWkQk2vfotIXnmsUMK6R

**Comment by elrodro83** [comment]
--validate

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamWorkQueueProcessingStrategyFactory.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  public void setWaitStrategy(String waitStrategy) {
+    this.waitStrategy = waitStrategy;
+  }
+
+  protected String getWaitStrategy() {
+    return waitStrategy;
+  }
+
+  protected Supplier<Scheduler> getRingBufferSchedulerSupplier(MuleContext muleContext, String schedulersNamePrefix) {
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamWorkQueueProcessingStrategyFactory.java]: Code Smell Code Smell: Remove this use of "MuleContext"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2vfotIXnmsUMK6S&open=AWkQk2vfotIXnmsUMK6S)

[ISSUE_KEY]: AWkQk2vfotIXnmsUMK6S

**Comment by elrodro83** [comment]
--validate

**Pull request merged**